src/accelerate/accelerator.py

```diff
@@ -1556,6 +1571,44 @@ def _prepare_megatron_lm(self, *args):
             )
         return tuple(result)
 
+    def _prepare_ipex(self, *args):
+        ipex_plugin = self.state.ipex_plugin
+        if not is_ipex_available():
+            raise ImportError(
+                "Using IPEX but IPEX is not installed or IPEX's version does not match current PyTorch, please refer"
+                " to https://github.com/intel/intel-extension-for-pytorch."
+            )
+        import intel_extension_for_pytorch as ipex
+
+        model = None
+        optimizer = None
+        dataloader = None
+        result = [obj for obj in args]
+        for obj in result:
+            if isinstance(obj, torch.utils.data.DataLoader) and dataloader is None:
+                dataloader = obj
+            elif isinstance(obj, torch.nn.Module):
+                model = obj
+            elif isinstance(obj, (torch.optim.Optimizer)):
+                optimizer = obj
+        if ipex_plugin.do_fusion and optimizer is None:
+            model.eval()
+            model = ipex.optimize(model, dtype=ipex_plugin.dtype, level="O1")
+            model = ipex_plugin.torch_jit_model_eval(model, dataloader, training=False)
+        else:
+            if not model.training:
+                model.train()
+            model, optimizer = ipex.optimize(
+                model, dtype=ipex_plugin.dtype, optimizer=optimizer, inplace=True, level="O1"
+            )
```

Comment:
This assumes that the user will always pass along the optimizer with the model when training it, and that when they don't pass an optimizer, they are doing evaluation only. As said above, many users prepare everything together. Those assumptions should be clearly documented as they break from the rest of the library.

Reply:
yes, we should not take the assumption like this. ipex will be only used for training part optimization as described above